--- lib/MasonX/Request/WithApacheSession.pm.orig	2006-03-28 14:33:16.000000000 -0500
+++ lib/MasonX/Request/WithApacheSession.pm	2006-03-28 14:36:06.000000000 -0500
@@ -89,7 +89,8 @@
     }
     elsif ( $self->can('cgi_object') )
     {
-        %extra = ( header_object => $self->cgi_object,
+        %extra = ( header_object => eval { $self->cgi_request } 
+                     || $self->cgi_object,
                    param_object  => $self->cgi_object,
                  );
     }
