--- lib/HTML/Mason/FakeApache.pm.orig	2006-03-28 14:11:10.000000000 -0500
+++ lib/HTML/Mason/FakeApache.pm	2006-03-28 14:13:46.000000000 -0500
@@ -65,6 +65,10 @@
 
 sub hostname { $_[0]->{query}->server_name }
 
+# CGI says "use this when using virtual hosts".  It falls
+# back to CGI->server_port.
+sub get_server_port { $_[0]->{query}->virtual_port }
+
 # Fake it by just giving the current time.
 sub request_time { time }
 
@@ -283,7 +287,7 @@
 # Do something with ErrorDocument?
 # sub custom_response {}
 
-# I think we'ev made this essentially the same thing.
+# I think we've made this essentially the same thing.
 BEGIN {
     local $^W;
     *send_cgi_header = \&send_http_header;
